s3.tf

```diff
@@ -89,26 +93,43 @@ resource "aws_s3_bucket_inventory" "apiary_bucket" {
 }
 
 resource "aws_s3_bucket_public_access_block" "apiary_bucket" {
-  count  = var.s3_block_public_access == true ? "${length(local.apiary_data_buckets)}" : 0
-  bucket = "${aws_s3_bucket.apiary_data_bucket.*.id[count.index]}"
+  for_each = var.s3_block_public_access == true ? {
+    for schema in local.schemas_info : "${schema["schema_name"]}" => schema
+  } : {}
+  bucket = aws_s3_bucket.apiary_data_bucket[each.key].id
 
   block_public_acls   = true
   block_public_policy = true
   ignore_public_acls  = true
 }
 
 resource "aws_s3_bucket_notification" "data_events" {
-  count  = "${var.enable_data_events == "" ? 0 : length(local.apiary_data_buckets)}"
-  bucket = "${aws_s3_bucket.apiary_data_bucket.*.id[count.index]}"
+  for_each = var.enable_data_events == "" ? {} : {
```

Comment:
probably can update `enable_data_events` to use `bool` data type

Reply:
Same question as above - migration script can't handle this for user.  Is this OK?

Reply:
same as above, can address in another PR